src/GitVersionCore/IncrementStrategyFinder.cs

```diff
@@ -0,0 +1,107 @@
+﻿namespace GitVersion
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using VersionCalculation.BaseVersionCalculators;
+    using LibGit2Sharp;
+
+    public enum CommitMessageVersionMode
```

Comment:
Although the PR submitter could still include merges in their own branch, so it is not a very hard restriction.